lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -413,10 +431,15 @@ def ensure(module, base, state, names):
                     # Environment is already uninstalled.
                     pass
 
-            installed = base.sack.query().installed()
-            for pkg_spec in pkg_specs:
-                if installed.filter(name=pkg_spec):
-                    base.remove(pkg_spec)
+            if _autoremove:
+                pkgs = base.sack.query().unneeded(base.sack, base._yumdb, debug_solver=base.conf.debug_solver)
```

Comment:
don't like such hacks, would be better if you will write RFE for DNF to make API for autoremove...